feat(smtp): write-only password support for SMTP credentials (#52) (#76)
hackthebox/terraform-provider-mailgun: docs/resources/smtp_credential.md

output "smtp_full_login" {
  value = mailgun_smtp_credential.app.full_login
}

# Write-only password (recommended). Requires Terraform CLI >= 1.11.
# The secret is never written to Terraform state. Bump password_wo_version to rotate.

> resource "random_password" "smtp" {

Reviewer: ```suggestion
ephemeral "random_password" "smtp" {
```

Author: This block is generated from the example file, so it now shows the ephemeral resource after `make generate` (fc26208).

### Optional

> **NOTE**: [Write-only arguments](https://developer.hashicorp.com/terraform/language/resources/ephemeral#write-only-arguments) are supported in Terraform 1.11 and later.

> - `password` (String, Sensitive, Deprecated) The password for SMTP authentication. This is write-only and cannot be read back from the API. Set this when creating a credential or when rotating the password of an imported credential. Leave it unset to keep the existing password of an imported credential.

Reviewer: ```suggestion
- `password` (String, Sensitive, Deprecated) The password for SMTP authentication. This is write-only and cannot be read back from the API. Set this when creating credentials or when rotating the password. Leave it unset to maintain the existing password during import.
```

Some suggestion on the text, _highly optional_

Author: Applied (at the schema `Description` source, then regenerated). fc26208.

> - `password_wo` (String, Sensitive, [Write-only](https://developer.hashicorp.com/terraform/language/resources/ephemeral#write-only-arguments)) Write-only password for SMTP authentication. The value is never stored in Terraform state. Set it together with password_wo_version and increment the version to rotate the password. Requires Terraform CLI >= 1.11.

Reviewer: ```suggestion
- `password_wo` (String, Sensitive, [Write-only](https://developer.hashicorp.com/terraform/language/resources/ephemeral#write-only-arguments)) Write-only password for SMTP authentication. The value is never stored in Terraform state. Always set it together with password_wo_version.
```
I think we can skip the password version specifics since we have a dedicated entry below

Author: Trimmed the version/CLI specifics from the `password_wo` description since `password_wo_version` has its own entry. fc26208.

> - `password_wo_version` (Number) Version counter for password_wo. Increment this value to rotate the write-only password. Required when password_wo is set.

Reviewer: ```suggestion
- `password_wo_version` (Number) Version counter for password_wo. Increment this value to rotate the write-only password. **Required** when password_wo is set.
```

Author: Bolded **Required** on `password_wo_version`. fc26208.
